internal/cloud_providers/aws/aws-ec2.go

```diff
@@ -0,0 +1,89 @@
+package cloudprovider
+
+import (
+	"github.com/RHEcosystemAppEng/cluster-iq/internal/inventory"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+// AWSEC2Connection represents the EC2 client for AWS
+type AWSEC2Connection struct {
+	client *ec2.EC2
+}
+
+// Creates a new EC2 client instance based on an AWSSession
+func NewAWSEC2Connection(session *session.Session) *AWSEC2Connection {
+	return &AWSEC2Connection{
+		client: ec2.New(session),
+	}
+}
+
+// WithEC2 configures an AWSConnection instance for including the EC2 client
+func WithEC2() AWSConnectionOption {
+	return func(conn *AWSConnection) {
+		conn.EC2 = NewAWSEC2Connection(conn.awsSession)
+	}
+}
+
+// GetRegionsList returns a list of the available AWS regions as a string array
+func (c *AWSEC2Connection) GetRegionsList() ([]string, error) {
+	// Getting regions from AWS API
+	regions, err := c.client.DescribeRegions(&ec2.DescribeRegionsInput{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Converting to string array
+	regionList := make([]string, 0)
+
+	for _, region := range regions.Regions {
+		regionList = append(regionList, *region.RegionName)
+	}
+
+	return regionList, nil
+}
+
+// GetInstances gets the list of EC2 instances and returns them as an Array if Inventory.Instances
+func (c *AWSEC2Connection) GetInstances() ([]inventory.Instance, error) {
+	// Getting Instances list
+	reservations, err := c.client.DescribeInstances(&ec2.DescribeInstancesInput{})
```

Comment:
Should be https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeInstancesPages like you did with Route53
No need to fix, just a todo